engine/src/cmd/collision.cpp

```diff
@@ -0,0 +1,263 @@
+        string fn( unit->filename );
+        string fac( FactionUtil::GetFaction( unit->faction ) );
+        unit->Kill();
+        _Universe->AccessCockpit()->savegame->AddUnitToSave( fn.c_str(), ENHANCEMENTPTR, fac.c_str(), reinterpret_cast<long>(unit));
+        apply_force = true;
+        return;
+    }
+}
+
+// This stops the unit from going through the other unit
+/*void Collision::applyForce(double elasticity, float& m2, Vector& v2)
+{
+    //for torque... location -- approximation hack of MR^2 for rotational inertia (moment of inertia currently just M)
+    Vector torque = force/(unit->radial_size*unit->radial_size);
+    Vector force  = this->force-torque; // Note the variable shadowing
+
+    float  max_force  = max_force_multiplier*(unit->limits.forward + unit->limits.retro + unit->limits.lateral + unit->limits.vertical);
+    float  max_torque = max_torque_multiplier*(unit->limits.yaw + unit->limits.pitch+unit->limits.roll);
+
+    //Convert from frames to seconds, so that the specified value is meaningful
+    max_force  = max_force / (unit->sim_atom_multiplier * simulation_atom_var);
+    max_torque = max_torque / (unit->sim_atom_multiplier * simulation_atom_var);
+    float torque_magnitude = torque.Magnitude();
+    float force_magnitude = force.Magnitude();
+    if (torque_magnitude > max_torque) {
+        torque *= (max_torque/torque_magnitude);
+    }
+    if (force_magnitude > max_force) {
+        force *= (max_force/force_magnitude);
+    }
+    unit->ApplyTorque( torque, location );
+    unit->ApplyForce( force-torque );
+}*/
+
+
+
+
+void Collision::dealDamage(Collision other_collision, float factor)
+{
+    if(!deal_damage)
+    {
+        return;
+    }
+
+    // Shorten variables for better visibility
+    float& m1 = mass;
+    float& m2 = other_collision.mass;
+    Vector& v1 = velocity;
+    Vector& v2 = other_collision.velocity;
+
+    float relative_velocity = (v1-v2).Magnitude();
+    float damage = relative_velocity * relative_velocity *
+            (m2)/(m1+m2) * factor;
+
+    unit->ApplyDamage(other_collision.location.Cast(),
+                      other_collision.normal,
+                      damage, unit, GFXColor( 1,1,1,2 ), other_collision.unit->owner
+                      != nullptr ? other_collision.unit->owner : this );
+}
+
+void Collision::applyForce(double elasticity, float& m2, Vector& v2)
+{
+    /*
+    The current implementation is of an imperfectly elastic collision.
+    An elastic collision is an encounter between two bodies in which the total kinetic energy of the two bodies remains the same. In an ideal, perfectly elastic collision, there is no net conversion of kinetic energy into other forms such as heat, noise, or potential energy.
+    https://en.wikipedia.org/wiki/Elastic_collision
+
+    This function adds an elasticity ratio to reduce velocity, due to heat, damage, etc.
+    */
+
+    // Ratio must be between 0 and 1
+    assert(elasticity>=0);
+    assert(elasticity<=1);
```

Comment:
I would not assert,  just log an error and do something reasonable (max, min, etc)

Reply:
Done.